server/accounts.go

```diff
@@ -3772,19 +3771,42 @@ func (dr *DirAccResolver) Fetch(name string) (string, error) {
 	} else {
 		dr.Lock()
 		srv := dr.Server
+		to := dr.fetchTimeout
 		dr.Unlock()
 		if srv == nil {
 			return "", err
 		}
-		return srv.fetch(dr, name) // lookup from other server
+		return srv.fetch(dr, name, to) // lookup from other server
 	}
 }
 
 func (dr *DirAccResolver) Store(name, jwt string) error {
 	return dr.saveIfNewer(name, jwt)
 }
 
-func NewDirAccResolver(path string, limit int64, syncInterval time.Duration, delete bool) (*DirAccResolver, error) {
+type dirResOption func(s *DirAccResolver) error
+
+// limits the amount of time spent waiting for an account fetch to complete
+func FetchTimeout(to time.Duration) dirResOption {
```

Comment:
Need to either make `fetchTimeout` private or `disResOption` public, since not recommended to return private types:

```
// golint
accounts.go:3790:37: exported func FetchTimeout returns unexported type server.dirResOption, which can be annoying to use
```


```suggestion
func fetchTimeout(to time.Duration) dirResOption {
```

Reply:
Since it's an argument to a public function that should probably stay public.

Reply:
fixed